Examples/Detectors/ContextualDetector/include/ActsExamples/ContextualDetector/AlignedDetector.hpp

```diff
@@ -63,21 +54,21 @@ class AlignedDetector {
 
     enum class Mode { Internal, External };
     Mode mode = Mode::Internal;
+
+    std::shared_ptr<const Acts::IMaterialDecorator> materialDecorator;
   };
 
-  std::pair<TrackingGeometryPtr, ContextDecorators> finalize(
-      const Config& cfg,
-      std::shared_ptr<const Acts::IMaterialDecorator> mdecorator);
+  explicit AlignedDetector(const Config& cfg);
 
-  std::vector<std::vector<std::shared_ptr<Generic::GenericDetectorElement>>>&
-  detectorStore() {
-    return m_detectorStore;
-  }
+  const DetectorStore& detectorStore() const { return m_detectorStore; }
 
  private:
+  Config m_cfg;
+
   /// The Store of the detector elements (lifetime: job)
-  std::vector<std::vector<std::shared_ptr<Generic::GenericDetectorElement>>>
-      m_detectorStore;
+  DetectorStore m_detectorStore;
```

Comment:
If this is here, why is it also in the base class then?

Reply:
This uses actually a different kind of store. I didn't look deeply in refactoring this